### PR TITLE
[WIP] Fix #1722: Added transition css class to appropriate elements

### LINKED ISSUE #1722: Make hover effect transitions consistent
Take a look at this GIF of me hovering over various UI elements.

hover

You'll notice the buttons in the toolbar change color immediately, while the menu color fades.

Let's make this consistent. I'd be curious what all of these look like with a fast transition time. Let's try ``.15s`` with an ``ease-out`` on the css transitions for these elements.

### PR DESCRIPTION
**Pull request by bkim49**
Added transition css to appropriate elements in userbar.less and editor.less to change transition time effectively. 

At the moment, the current transition time for both files are set to 0.15s. I can change the time back to 0.5s or whatever you think is the best transition time @flukeout @gideonthomas.

### CODE REVIEW

**Comment by flukeout** [comment]
Checking now!

File: public/editor/stylesheets/editor.less [hunk 1]
```diff
@@ -258,6 +260,7 @@ body {
     width: 15px;
     position: relative;
     top: 0;
+	.transition;
```
Comment (flukeout) [inline public/editor/stylesheets/editor.less]: Please keep spacing / indentation consistent. I think you're using Tabs while the source file is using spaces.

**Comment by flukeout** [comment]
Looks good overall! Couple of things: 

* Please fix the spacing / indentation (commented on one of the lines where this happens)
* Please add an ``ease-out`` timing to the transitions.

I think you can add it to the each of the ``.transition {`` definitions in ``userbar.less`` and ``editor.less`` - currently there is no timing function defined.

Thanks!

**Comment by bkim49** [comment]
Sorry for late reply!

I have changed the spacing and added transition timing function as well. I tried to look at the difference of having transition timing function as ease out or ease in etc, but I could not really tell a  difference. 

@flukeout could you let me know if the changes were correct?

**Comment by flukeout** [comment]
Will review a little later today, thank you!

File: public/editor/stylesheets/editor.less [hunk 2]
```diff
@@ -850,13 +850,14 @@ body {
...
-    -o-transition: @transitionTime;
-    transition: @transitionTime;
+    -webkit-transition: @transitionTime ease-out;
+    -moz-transition: @transitionTime ease-out;
+    -ms-transition: @transitionTime ease-out;
+    -o-transition: @transitionTime ease-out;
+    transition: @transitionTime ease-out;
     -webkit-backface-visibility: hidden;
     backface-visibility: hidden;
+	transition-timing-function: ease-out;
```
Comment (Pomax) [inline public/editor/stylesheets/editor.less]: looks like indentation went wrong here again

File: public/resources/stylesheets/userbar.less [hunk 3]
```diff
@@ -255,19 +258,21 @@ body {
   -ms-transition: opacity @transitionTime ease-in-out;
   -o-transition: opacity @transitionTime ease-in-out;
   transition: opacity @transitionTime ease-in-out;
+  transition-timing-function: opacity @transitionTime ease-out;
```
Comment (flukeout) [inline public/resources/stylesheets/userbar.less]: Was this line added or changed by mistake? It's incorrect.

**Comment by bkim49** [comment]
@Pomax thank you for letting me know!

@flukeout omg. I was testing stuff and forgot to remove them. I should really double check my work. Ill fix this as well. Thank you.

**Comment by humphd** [comment]
@bkim49 np, we all leave things in our commits.  One tip: when you go to submit a PR, always go and read the `Files changed` tab to see what's being sent, in case you missed something.

**Comment by bkim49** [comment]
@humphd I will remeber that. Thank you.

**Comment by flukeout** [comment]
@bkim49 Just want to check in on this PR - will you be updating it with the minor fixes?

**Comment by bkim49** [comment]
@flukeout I have fixed the minor error on my part. Sorry for taking so long for this fix!

File: public/editor/stylesheets/editor.less (hunk 2 above)
Comment (gideonthomas) [inline public/editor/stylesheets/editor.less]: @bkim49 I think you're using tabs vs. spaces. We use 2-space indents throughout the code so would you be able to convert this to 4 spaces?

**Comment by humphd** [comment]
@bkim49 looks like this will need a rebase for changes in `public/editor/stylesheets/editor.less`.  Have you made all the other changes?

**Comment by bkim49** [comment]
@flukeout @Pomax I have made changes to the requests!

**Comment by flukeout** [comment]
Hi @bkim49 not sure if you noticed but the indentation still needs to be fixed on ``line 872`` in ``editor.less`` - you can see the comments if you click on the  Files Changed tab above and scroll down to see the comments. Almost there!

**Comment by flukeout** [comment]
Ok - testing it out again!

**Comment by flukeout** [comment]
Thanks for your work on this @bkim49 !

**Pull request merged**